pest/src/parser_state.rs

```diff
@@ -86,7 +86,8 @@ where
                     positives: state.pos_attempts.clone(),
                     negatives: state.neg_attempts.clone()
                 },
-                unsafe { position::new(input.as_bytes(), state.attempt_pos) }
+                // TODO(performance): Guarantee state.attempt_pos is a valid position
```

Comment:
This one is me.

I mean this TODO to be guarantee as in go through and find the manipulators of attempt_pos and make sure that all of them are making sure to set it to a character boundary, which I haven't done. That's why I changed it to the safe implementation rather than the unsafe one for now.

Honestly, this is probably cold enough that just using the unwrap is fine.